src/test/java/com/atlassian/db/replica/api/AuroraClusterDiscoveryTest.java

```diff
@@ -74,6 +74,19 @@ void shouldDiscoverReplicaConnectionsWhenScaledDown() {
         assertThat(replicas).hasSize(1);
     }
 
+    @Test
+    void shouldCloseDatabaseConnectionAfterDiscoveryReplicaConnections() throws SQLException {
```

Comment:
Interesting. I think it shouldn't. DualConnection closes the connection if the replica is not consistent, and keeps it if consistent. I think closing the connection breaks a hidden contract. It also shouldn't be a problem for a Waiter as it also closes the connection.

Reply:
Oh. I see ReplicaConsistency doesn't close the connection. AuroraClusterDiscovery does it. Seems to be 🆗.

Reply:
Would be great to verify all of those in tests.